README.md

```diff
@@ -1,30 +1,135 @@
 # PostCSS Customprop Validate
 
-[PostCSS] plugin to validate default values of CSS custom properties.
+[PostCSS] plugin to validate fallback values of CSS custom properties.
 
-[PostCSS]: https://github.com/postcss/postcss
+Validating the fallback values of CSS custom properties manually can be error prone and challening if they are huge in number. This plugin removes this effort altogether.
+
+The plugin accepts an object containing custom properties in plugin [options]. Using it as the source of truth, the plugin validates the CSS and finds out the custom properties with incorrect fallback values.
+
+The plugin can also modify the source file with the updated CSS -- with correct fallback values. See `write` under [options]
+
+> If you use a code formatter like [prettier], you may need to re-run it after the plugin modifies the source file.
+
+## Options
+
+The plugin accepts 3 options:
+
+**`properties`**
+
+An object containg key and fallback values of custom properties
+
+| type   | required | default |
+| ------ | -------- | ------- |
+| Object | true     | N/A     |
+
+**`write`**
+
+Writes the source CSS file with the updated CSS
+
+| type    | required | default |
+| ------- | -------- | ------- |
+| boolean | false    | false   |
+
+**`callback`**
+
+Callback to handle the response of the plugin. The callback accepts 2 arguments:
+
+- error: an error object
+- data: the response
+
+| type     | required | default  |
+| -------- | -------- | -------- |
+| Function | false    | () => {} |
+
+## Response
+
+The response is a list of object of custom properties with incorrect fallback values. The plugin's result also contains the response -- `result.wrongProps`. Shape of the object:
+
+- `path` {string} - absolute path of the CSS file
+- `line` {number} - line number containing the custom property
+- `key` {string} - key of the custom property
+- `current` {string} - current fallback value
+- `expexted` {string} - expected fallback value
+
+## Example
+
+Sample `properties` option:
```

Comment:
* As I understand, this is the input data to the function. Should not call it option? Options are flags that control the behavior of the function — generally speaking.

Reply:
PostCSS calls them plugin options. Hence, I added the term _options_ to remind the user that `properties` is a plugin option.